src/mem/alloc.h

```diff
@@ -473,13 +473,13 @@ namespace snmalloc
     }
 
   public:
-    SNMALLOC_FAST_PATH static size_t alloc_size(const void* p)
+    SNMALLOC_FAST_PATH size_t alloc_size(const void* p)
     {
 #ifdef SNMALLOC_PASS_THROUGH
       return external_alloc::malloc_usable_size(const_cast<void*>(p));
 #else
       // This must be called on an external pointer.
-      size_t size = ChunkMap::get(address_cast(p));
```

Comment:
I, very belatedly, took some time to look at this in more detail.  Clang's able, if both `alloc_size` and `external_pointer` are de-`static`-ed, to generate almost the same code, differing just in register selection, for the `perf/external_pointer` test.  (Oddly, changing just one makes it perform worse, but since we're not doing that...)